### ISSUE TITLE
Tycho build NoSuchMethodError with xbase.lib 2.10

### ISSUE THREAD

**Issue by gauravk08** [eclipse/xtext-xtend#148]
I am start getting tycho build error from yesterday 01-Feb-2017. 

Exception-

# ## Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default of goal org.eclipse.xtend:xtend-maven-plugin:2.10.0:compile failed: An API incompatibility was encountered while executing org.eclipse.xtend:xtend-maven-plugin:2.10.0:compile: java.lang.NoSuchMethodError: org.eclipse.xtend2.lib.StringConcatenation.append(Ljava/lang/String;)

Builds was working fine before 01-Feb-2017 afternoon , and stop working after that.

Do I have to get any plugin to updated or something I am missing. I am using JRE1.7 to build.

Tycho setup, related to xtend -
<properties>		<xtext.xtend.url>http://download.eclipse.org/modeling/tmf/xtext/updates/composite/releases/</xtext.xtend.url>
<xtend.version>2.10.0</xtend.version>
</properties>

<dependencies>
		<dependency>
			<groupId>org.eclipse.xtend</groupId>
			<artifactId>org.eclipse.xtend.lib</artifactId>
			<version>${xtend.version}</version>
		</dependency>
		<dependency>
			<groupId>org.eclipse.xtend</groupId>
			<artifactId>org.eclipse.xtend.lib.macro</artifactId>
			<version>${xtend.version}</version>
		</dependency>
	</dependencies>
<build>
        <plugins>
          <plugin>
            <groupId>org.eclipse.tycho</groupId>
            <artifactId>tycho-maven-plugin</artifactId>
            <version>${tycho.version}</version>
            <extensions>true</extensions>
          </plugin>
          <plugin>
            <groupId>org.eclipse.tycho</groupId>
            <artifactId>target-platform-configuration</artifactId>
            <version>${tycho.version}</version>
            <configuration>
              <environments>
                <environment>
                  <os>win32</os>
                  <ws>win32</ws>
                  <arch>x86</arch>
                </environment>
              </environments>
            </configuration>
          </plugin>
          <plugin>
            <groupId>org.eclipse.xtend</groupId>
            <artifactId>xtend-maven-plugin</artifactId>
            <version>${xtend.version}</version>
            <executions>
              <execution>
                <goals>
                  <goal>compile</goal>
                  <goal>testCompile</goal>
                  <goal>xtend-install-debug-info</goal>
                  <goal>xtend-test-install-debug-info</goal>
                </goals>
              </execution>
            </executions>
            <configuration>
              <outputDirectory>xtend-gen</outputDirectory>
              <testOutputDirectory>${project.build.directory}/xtend-gen/test</testOutputDirectory>
              <encoding>${project.build.sourceEncoding}</encoding>
            </configuration>
          </plugin>
          <plugin>
            <artifactId>maven-compiler-plugin</artifactId>
            <version>3.3</version>
            <configuration>
[Xtend_XBASE_Lib_Issue_Log.txt](https://github.com/eclipse/xtext-xtend/files/747403/Xtend_XBASE_Lib_Issue_Log.txt)

              <source>1.6</source>
              <target>1.6</target>
              <encoding>${project.build.sourceEncoding}</encoding>
            </configuration>
          </plugin>
        </plugins>
      </build>

**Comment by svenefftinge** [comment]
In 2.11 the xtend compiler generates code that makes use of a new api in StringConcatenation. 
This error happens if you generate with xtend 2.11 and compile against xbase.lib 2.10.

**Comment by gauravk08** [comment]
@svenefftinge  thanks for reply, 
But we are using tycho maven for xtend compiler and using same version 2.10.0 everywhere.
Seems something wrong while maven compilation.

Here is jenkins log where we are using only 2.10.0 xtend only, or I missed something.

***********************************************************************************
realm =    plugin>org.eclipse.xtend:xtend-maven-plugin:2.10.0
strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
urls[0] = file:/var/lib/jenkins/.m2/org/eclipse/xtend/xtend-maven-plugin/2.10.0/xtend-maven-plugin-2.10.0.jar
urls[1] = file:/var/lib/jenkins/.m2/com/google/guava/guava/19.0-rc3/guava-19.0-rc3.jar
urls[2] = file:/var/lib/jenkins/.m2/org/eclipse/xtend/org.eclipse.xtend.core/2.10.0/org.eclipse.xtend.core-2.10.0.jar
urls[3] = file:/var/lib/jenkins/.m2/org/eclipse/xtext/org.eclipse.xtext.xbase/2.10.0/org.eclipse.xtext.xbase-2.10.0.jar
urls[4] = file:/var/lib/jenkins/.m2/org/eclipse/xtext/org.eclipse.xtext/2.10.0/org.eclipse.xtext-2.10.0.jar
urls[5] = file:/var/lib/jenkins/.m2/org/eclipse/equinox/org.eclipse.equinox.common/3.6.0/org.eclipse.equinox.common-3.6.0.jar
urls[6] = file:/var/lib/jenkins/.m2/org/eclipse/xtext/org.eclipse.xtext.common.types/2.10.0/org.eclipse.xtext.common.types-2.10.0.jar
urls[7] = file:/var/lib/jenkins/.m2/org/eclipse/emf/org.eclipse.emf.mwe2.lib/2.8.3/org.eclipse.emf.mwe2.lib-2.8.3.jar
urls[8] = file:/var/lib/jenkins/.m2/org/eclipse/emf/org.eclipse.emf.mwe2.runtime/2.8.3/org.eclipse.emf.mwe2.runtime-2.8.3.jar
urls[9] = file:/var/lib/jenkins/.m2/org/eclipse/xtext/org.eclipse.xtext.smap/2.10.0/org.eclipse.xtext.smap-2.10.0.jar
urls[10] = file:/var/lib/jenkins/.m2/org/eclipse/xtext/org.eclipse.xtext.xbase.junit/2.10.0/org.eclipse.xtext.xbase.junit-2.10.0.jar
urls[11] = file:/var/lib/jenkins/.m2/org/eclipse/xtext/org.eclipse.xtext.junit4/2.10.0/org.eclipse.xtext.junit4-2.10.0.jar
urls[12] = file:/var/lib/jenkins/.m2/junit/junit/4.8.1/junit-4.8.1.jar
urls[13] = file:/var/lib/jenkins/.m2/org/eclipse/jdt/org.eclipse.jdt.core/3.10.0/org.eclipse.jdt.core-3.10.0.jar
urls[14] = file:/var/lib/jenkins/.m2/org/eclipse/core/org.eclipse.core.resources/3.7.100/org.eclipse.core.resources-3.7.100.jar
urls[15] = file:/var/lib/jenkins/.m2/org/eclipse/core/org.eclipse.core.expressions/3.4.300/org.eclipse.core.expressions-3.4.300.jar
urls[16] = file:/var/lib/jenkins/.m2/org/eclipse/core/org.eclipse.core.runtime/3.7.0/org.eclipse.core.runtime-3.7.0.jar
urls[17] = file:/var/lib/jenkins/.m2/org/eclipse/osgi/org.eclipse.osgi/3.7.1/org.eclipse.osgi-3.7.1.jar
urls[18] = file:/var/lib/jenkins/.m2/org/eclipse/core/org.eclipse.core.jobs/3.5.100/org.eclipse.core.jobs-3.5.100.jar
urls[19] = file:/var/lib/jenkins/.m2/org/eclipse/equinox/org.eclipse.equinox.registry/3.5.101/org.eclipse.equinox.registry-3.5.101.jar
urls[20] = file:/var/lib/jenkins/.m2/org/eclipse/equinox/org.eclipse.equinox.preferences/3.4.1/org.eclipse.equinox.preferences-3.4.1.jar
urls[21] = file:/var/lib/jenkins/.m2/org/eclipse/core/org.eclipse.core.contenttype/3.4.100/org.eclipse.core.contenttype-3.4.100.jar
urls[22] = file:/var/lib/jenkins/.m2/org/eclipse/equinox/org.eclipse.equinox.app/1.3.100/org.eclipse.equinox.app-1.3.100.jar
urls[23] = file:/var/lib/jenkins/.m2/org/eclipse/core/org.eclipse.core.filesystem/1.3.100/org.eclipse.core.filesystem-1.3.100.jar
urls[24] = file:/var/lib/jenkins/.m2/org/eclipse/text/org.eclipse.text/3.5.101/org.eclipse.text-3.5.101.jar
urls[25] = file:/var/lib/jenkins/.m2/org/eclipse/core/org.eclipse.core.commands/3.6.0/org.eclipse.core.commands-3.6.0.jar
urls[26] = file:/var/lib/jenkins/.m2/org/eclipse/xtext/org.eclipse.xtext.xbase.lib/2.10.0/org.eclipse.xtext.xbase.lib-2.10.0.jar
urls[27] = file:/var/lib/jenkins/.m2/org/eclipse/xtext/org.eclipse.xtext.util/2.10.0/org.eclipse.xtext.util-2.10.0.jar
urls[28] = file:/var/lib/jenkins/.m2/org/eclipse/xtend/org.eclipse.xtend.lib/2.10.0/org.eclipse.xtend.lib-2.10.0.jar
urls[29] = file:/var/lib/jenkins/.m2/org/eclipse/xtend/org.eclipse.xtend.lib.macro/2.10.0/org.eclipse.xtend.lib.macro-2.10.0.jar
urls[30] = file:/var/lib/jenkins/.m2/org/eclipse/emf/org.eclipse.emf.common/2.12.0/org.eclipse.emf.common-2.12.0.jar
urls[31] = file:/var/lib/jenkins/.m2/org/eclipse/emf/org.eclipse.emf.ecore/2.12.0/org.eclipse.emf.ecore-2.12.0.jar
urls[32] = file:/var/lib/jenkins/.m2/com/google/inject/guice/3.0/guice-3.0.jar
urls[33] = file:/var/lib/jenkins/.m2/javax/inject/javax.inject/1/javax.inject-1.jar
urls[34] = file:/var/lib/jenkins/.m2/aopalliance/aopalliance/1.0/aopalliance-1.0.jar
urls[35] = file:/var/lib/jenkins/.m2/log4j/log4j/1.2.16/log4j-1.2.16.jar
urls[36] = file:/var/lib/jenkins/.m2/org/ow2/asm/asm-commons/5.0.1/asm-commons-5.0.1.jar
urls[37] = file:/var/lib/jenkins/.m2/org/ow2/asm/asm-tree/5.0.1/asm-tree-5.0.1.jar
urls[38] = file:/var/lib/jenkins/.m2/org/ow2/asm/asm/5.0.1/asm-5.0.1.jar
urls[39] = file:/var/lib/jenkins/.m2/org/eclipse/aether/aether-util/1.0.0.v20140518/aether-util-1.0.0.v20140518.jar
urls[40] = file:/var/lib/jenkins/.m2/javax/enterprise/cdi-api/1.0/cdi-api-1.0.jar
urls[41] = file:/var/lib/jenkins/.m2/javax/annotation/jsr250-api/1.0/jsr250-api-1.0.jar
urls[42] = file:/var/lib/jenkins/.m2/org/eclipse/sisu/org.eclipse.sisu.inject/0.3.0.M1/org.eclipse.sisu.inject-0.3.0.M1.jar
urls[43] = file:/var/lib/jenkins/.m2/org/sonatype/sisu/sisu-guice/3.2.3/sisu-guice-3.2.3-no_aop.jar
urls[44] = file:/var/lib/jenkins/.m2/org/codehaus/plexus/plexus-interpolation/1.21/plexus-interpolation-1.21.jar
urls[45] = file:/var/lib/jenkins/.m2/org/codehaus/plexus/plexus-utils/3.0.20/plexus-utils-3.0.20.jar
urls[46] = file:/var/lib/jenkins/.m2/org/codehaus/plexus/plexus-component-annotations/1.5.5/plexus-component-annotations-1.5.5.jar
urls[47] = file:/var/lib/jenkins/.m2/org/sonatype/plexus/plexus-sec-dispatcher/1.3/plexus-sec-dispatcher-1.3.jar
urls[48] = file:/var/lib/jenkins/.m2/org/sonatype/plexus/plexus-cipher/1.4/plexus-cipher-1.4.jar
urls[49] = file:/var/lib/jenkins/.m2/org/eclipse/emf/org.eclipse.emf.ecore.xmi/2.12.0/org.eclipse.emf.ecore.xmi-2.12.0.jar
urls[50] = file:/var/lib/jenkins/.m2/org/antlr/antlr-runtime/3.2/antlr-runtime-3.2.jar
urls[51] = file:/var/lib/jenkins/.m2/org/eclipse/emf/org.eclipse.emf.codegen/2.11.0-v20150806-0404/org.eclipse.emf.codegen-2.11.0-v20150806-0404.jar
*************************************************************************************************************

**Comment by cdietrich** [comment]
which xtext version do you build against / is in your tycho target platform?

**Comment by gauravk08** [comment]
We are tying to build with Xtext- 2.10.0 and using our tycho-0.24.0

**Comment by cdietrich** [comment]
do you restrict the xtext version of your bundles to 

```
Require-Bundle: org.eclipse.xtext;bundle-version="[2.10.0,2.11.0)",
```

or via beeing selective in your target platform and/or pom

**Comment by cdietrich** [comment]
p.s. having a small reproducible example would be nice

**Comment by cdietrich** [comment]
@gauravk08 any news on this?

**Comment by gauravk08** [comment]
Hi Christian,
I tried number of attempts  to fix this issue including fix xtext version through Require-Bundle: org.eclipse.xtext;bundle-version="[2.10.0,2.11.0)", but it not worked for me.  It was bit urgent to fix as our daily build was failing so I created my own local update site with 2.10.0 and point this to parent pom.xml and it is working fine for me. I have not investigated much after that. But still curious to know why this filed.
